pkg/storage/image.go

```diff
@@ -556,6 +558,9 @@ func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
 	domain, remainder := splitDockerDomain(imageName)
 	if domain != "" {
 		// this means the image is already fully qualified
+		if svc.blockedRegistries[domain] {
```

Comment:
An alternative to that would be `sysregistriesv2.FindRegistry(...)` and check for reg.Blocked (nil == not blocked).

Reply:
Yes please; use the API to do lookups (e.g. to allow the package to optimize them in the future).

Reply:
`FindRegistry`